### PR TITLE
Add TFM guidance for Aspire Client Library Integrations

### PR DESCRIPTION
**Pull request by eerhardt**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/aspire/pull/5376)

### CODE REVIEW

**Review by DamianEdwards — APPROVED** [review]
Looks good

File: src/Components/README.md [hunk 1]
```diff
@@ -20,6 +20,17 @@ In summary we encourage and are excited to accept contributions of components, b
...
 Each component is in its own NuGet package, and can version independently, including declaring itself in a preview state using the standard SemVer and NuGet mechanisms. However we expect the major and minor version of components to follow the core Aspire packages to make it easier to reason about dependencies. We expect to typically push updates to all components at the same time we update the core Aspire packages, but we have the ability to push an updated component at any other time if necessary, for example where changes to the underlying client library makes it necessary.
 
+### Target Framework(s)
+
+The Aspire component must support the [latest LTS version of .NET](https://dotnet.microsoft.com/platform/support/policy/dotnet-core) and may optionally support a higher STS version, if one exists. For example:
+
+| .NET Aspire Version | Targets                         |
+|---------------------|---------------------------------|
+| 8.x                 | net8.0                          |
+| 9.x                 | net8.0 (optionally  + `net9.0`) |
```
Comment (sebastienros) [inline src/Components/README.md:30]: I find it confusing, does "optionally" apply to net8.0 or net9.0

Comment (eerhardt) [reply src/Components/README.md]: Are the new changes better?

File: src/Components/README.md [hunk 2]
```diff
@@ -20,6 +20,17 @@ In summary we encourage and are excited to accept contributions of components, b
...
+| 10.x                | net10.0                         |
+| 11.x                | net10.0 (optionally + `net11.0`)|
```
Comment (sebastienros) [inline src/Components/README.md]:
```suggestion
| .NET Aspire Version | Targets                            |
|---------------------|------------------------------------|
| 8.x                 | net8.0                             |
| 9.x                 | net8.0 (and optionally `net9.0`)   |
| 10.x                | net10.0                            |
| 11.x                | net10.0 (and optionally `net11.0`) |
```

**Review by sebastienros — APPROVED** [review]

**Pull request merged**